packages/driver/cypress/integration/e2e/multi-domain/multi_domain_yield_spec.ts

```diff
@@ -0,0 +1,238 @@
+      .get('[data-cy="dom-check"]')
+      .invoke('text')
+
+      const p = new Promise((resolve, reject) => {
+        setTimeout(() => {
+          resolve('text')
+        }, 50)
+      })
+
+      return p
+    }).should('equal', 'From a secondary domain')
+  })
+
+  it('errors if a cy command is present and it returns a sync value', (done) => {
+    cy.on('fail', (err) => {
+      assertLogLength(logs, 6)
+      expect(logs[5].get('error')).to.eq(err)
+      expect(err.message).to.include('`cy.switchToDomain()` failed because you are mixing up async and sync code.')
+
+      done()
+    })
+
+    cy.switchToDomain('foobar.com', () => {
+      cy
+      .get('[data-cy="dom-check"]')
+      .invoke('text')
+
+      return 'text'
+    })
+  })
+
+  it('yields synchronously', () => {
+    cy.switchToDomain('foobar.com', () => {
+      return 'From a secondary domain'
+    }).should('equal', 'From a secondary domain')
+  })
+
+  it('yields asynchronously', () => {
+    cy.switchToDomain('foobar.com', async () => {
+      return new Promise((resolve: (val: string) => any, reject) => {
+        setTimeout(() => {
+          resolve('From a secondary domain')
+        }, 1000)
+      })
+    }).should('equal', 'From a secondary domain')
+  })
+
+  it('succeeds if subject cannot be serialized and is not accessed synchronously', () => {
+    cy.switchToDomain('foobar.com', () => {
+      return {
+        symbol: Symbol(''),
+      }
+    }).then((obj) => {
+      return 'object not accessed'
+    }).should('equal', 'object not accessed')
+  })
+
+  it('throws if subject cannot be serialized and is accessed synchronously', (done) => {
+    cy.on('fail', (err) => {
+      assertLogLength(logs, 6)
+      expect(logs[5].get('error')).to.eq(err)
+      expect(err.message).to.include('`cy.switchToDomain()` could not serialize the subject due to one of it\'s properties not being supported by the structured clone algorithm.')
+
+      done()
+    })
+
+    cy.switchToDomain('foobar.com', () => {
+      return {
+        symbol: Symbol(''),
+      }
+    }).then((obj) => {
+      // This will fail accessing the symbol
+      // @ts-ignore
+      return obj.symbol
+    })
+  })
+
+  it('succeeds if subject cannot be serialized and is not accessed', () => {
+    cy.switchToDomain('foobar.com', () => {
+      cy
+      .get('[data-cy="dom-check"]')
+    }).then((obj) => {
+      return 'object not accessed'
+    }).should('equal', 'object not accessed')
+  })
+
+  it('throws if subject cannot be serialized and is accessed', (done) => {
+    cy.on('fail', (err) => {
+      assertLogLength(logs, 7)
+      expect(logs[6].get('error')).to.eq(err)
+      expect(err.message).to.include('`cy.switchToDomain()` could not serialize the subject due to one of it\'s properties not being supported by the structured clone algorithm.')
+
+      done()
+    })
+
+    cy.switchToDomain('foobar.com', () => {
+      cy
+      .get('[data-cy="dom-check"]')
+    }).should('equal')
```

Comment:
This seems incomplete since it doesn't have an expected parameter. As a side note, when I see these `should`s I think it is going to be a valid assertion.

Reply:
it is a malformed should, i'll add another parameter, but the mere act of calling the .should will cause the proxy to throw an error.

Reply:
When I see the should I'm assuming that it's a valid assertion and something you are expecting to pass. I feels odd to me to be using the `should` to invoke the subject to cause an error.

Reply:
I've update the test to look more realistic, but regardless simply hanging a .should off a switchToDomain call when the subject failed to serialize will cause an error to throw.